### ISSUE TITLE
RHEL 7.3 as VM OS

### ISSUE THREAD

**Issue by yaron2** [Azure/acs-engine#341]
Currently all VMs are deployed with Ubuntu.
I'd like to be able to customize the template to use RHEL 7.3.

I can do it myself and contribute back and would be happy to get pointers and guidelines for what needs to be done.

**Comment by yaron2** [comment]
RHEL images on Azure do not support cloud-init..

**Comment by rgardler** [comment]
Sorry, closed wrong issue :-(

**Comment by ams0** [comment]
I started working on CentOS for this and hit a few blockers. I used both Microsoft.OSTCExtensions and Microsoft.Azure.Extenstions for Swarm mode, but it seems that the script in https://github.com/ams0/acs-engine/blob/master/parts/configure-swarmmode-cluster.sh doesn't return, leaving the VMSS in perpetual creating status. I haven't tried with Kubernetes, but it seems way harder, as we also need to pass the whole certificate tree.

I'm still pursuing this and I'll report if I make progress.
